### PR TITLE
Ability to remove 2FA from account login + payouts 2FA open in new tab

### PR DESCRIPTION
**Pull request by sbinlondon**
Relates https://github.com/opencollective/opencollective/issues/3711

Fixes https://github.com/opencollective/opencollective/issues/3722

* Adds ability to remove 2FA from an account's login process if they don't want it anymore.
* For 2FA payouts, if a user doesn't have 2FA set up yet, it opens the profile settings in a new tab and goes directly to the 2FA settings instead of opening the general profile settings menu in the same tab

Screenshot 2020-12-01 at 14 32 29
Screenshot 2020-12-01 at 14 32 55
Screenshot 2020-12-01 at 14 33 05

### CODE REVIEW

File: components/edit-collective/sections/SetupTwoFactorAuth.js [hunk 1]
```diff
@@ -175,17 +203,114 @@ class SetupTwoFactorAuth extends React.Component {
...
+                            </MessageBox>
+                            {disableError && (
+                              <MessageBox type="error" withIcon mb={3}>
+                                {disableError}
+                              </MessageBox>
+                            )}
+                            <P>
+                              <FormattedMessage
+                                id="TwoFactorAuth.Disable.Info"
+                                defaultMessage="If you would like to remove 2FA from your account, you will need to enter the code from your authenticator app on more time."
```
Comment (SudharakaP) [inline components/edit-collective/sections/SetupTwoFactorAuth.js:267]: @sbinlondon : Minor typo; .... from your authenticator app **one** more time.

File: lang/ca.json [hunk 2]
```diff
@@ -1675,7 +1675,12 @@
   "transferwise.error.insufficientFunds": "Not have enough funds in your {currency} balance. Please top up your account and try again.",
   "transferwise.error.notConnected": "Host is not connected to Transferwise",
   "tweetIt": "Tweet it",
-  "TwoFactorAuth.Setup.AlreadyAdded": "Two-factor authentication is enabled on this account. Well done! 🎉",
+  "TwoFactorAuth": "Two-factor authentication",
+  "TwoFactorAuth.Disable.Button": "Disable 2FA",
+  "TwoFactorAuth.Disable.Header": "Are you sure you want to remove two-factor authentication from your account?",
+  "TwoFactorAuth.Disable.Info": "If you would like to remove 2FA from your account, you will need to enter the code from your authenticator app on more time.",
+  "TwoFactorAuth.Disable.Warning": "Removing 2FA from your account can make it less secure.",
```
Comment (SudharakaP) [inline lang/ca.json:1682]: @sbinlondon : Same typo in all lang files as above; "... **one** more time."

File: components/edit-collective/sections/SetupTwoFactorAuth.js [hunk 3]
```diff
@@ -175,17 +203,114 @@ class SetupTwoFactorAuth extends React.Component {
...
+                                    autoFocus
+                                  />
+                                )}
+                              </StyledInputField>
+                            </Form>
+                          </ModalBody>
+                          <ModalFooter>
+                            <Container display="flex" justifyContent="flex-end">
+                              <StyledButton mx={20} onClick={() => this.setState({ disablingTwoFactorAuth: false })}>
+                                <FormattedMessage id="actions.cancel" defaultMessage={'Cancel'} />
```
Comment (znarf) [inline components/edit-collective/sections/SetupTwoFactorAuth.js:300]: Why note just quotes here and below?

File: components/edit-collective/sections/SetupTwoFactorAuth.js [hunk 4]
```diff
@@ -307,11 +432,27 @@ const twoFactorAuthToAccountMutation = gqlV2/* GraphQL */ `
   }
 `;
 
+const twoFactorAuthFromAccountMutation = gqlV2/* GraphQL */ `
```
Comment (znarf) [inline components/edit-collective/sections/SetupTwoFactorAuth.js:435]: It should be clear it's `removeTwoFactorAuthFromAccount`

File: components/edit-collective/sections/SetupTwoFactorAuth.js [hunk 5]
```diff
@@ -339,6 +480,10 @@ const addAccountHasTwoFactorAuthData = graphql(accountHasTwoFactorAuthQuery, {
   }),
 });
 
-const addGraphql = compose(addTwoFactorAuthToAccountMutation, addAccountHasTwoFactorAuthData);
+const addGraphql = compose(
+  addTwoFactorAuthToAccountMutation,
+  removeTwoFactorAuthFromAccountMutation,
```
Comment (znarf) [inline components/edit-collective/sections/SetupTwoFactorAuth.js:485]: Usually, in our common practices, the `add` is not about adding or removing something to the account. It's about adding the mutation to the component. To follow practices in the project it should be:

`addAddTwoFactorAuthToAccountMutation`
`addRemoveTwoFactorAuthToAccountMutation`

If that sounds stupid, you can avoid it using an alternative syntax:

```
const addGraphql = compose(
  graphql(addTwoFactorAuthFromAccountMutation, {
    name: 'removeTwoFactorAuthTokenFromIndividual',
    options: { context: API_V2_CONTEXT },
  }),
  ...
);
```

File: components/edit-collective/sections/SetupTwoFactorAuth.js [hunk 6]
```diff
@@ -307,11 +432,27 @@ const twoFactorAuthToAccountMutation = gqlV2/* GraphQL */ `
...
+  }
+`;
+
 const addTwoFactorAuthToAccountMutation = graphql(twoFactorAuthToAccountMutation, {
   name: 'addTwoFactorAuthTokenToIndividual',
   options: { context: API_V2_CONTEXT },
 });
 
+const removeTwoFactorAuthFromAccountMutation = graphql(twoFactorAuthFromAccountMutation, {
+  name: 'removeTwoFactorAuthTokenFromIndividual',
```
Comment (znarf) [inline components/edit-collective/sections/SetupTwoFactorAuth.js:452]: It's confusing that in one case we "removeTwoFactorAuthFromAccount" and in the other "removeTwoFactorAuthTokenFromIndividual". Simply keep "removeTwoFactorAuthFromAccount" for the name.

**Review by znarf — CHANGES_REQUESTED** [review]
Looks good at first glance, relatively minor comments.

Can you share screenshots so we see how it's looking in the issue?

File: components/edit-collective/sections/SetupTwoFactorAuth.js (hunk 6 above)
Comment (sbinlondon) [reply components/edit-collective/sections/SetupTwoFactorAuth.js:452]: That would require an API change. I'll change them all to 'individual' to keep it consistent on the frontend.

**Comment by sbinlondon** [comment]
> Can you share screenshots so we see how it's looking in the issue?

Done!

File: components/edit-collective/sections/SetupTwoFactorAuth.js (hunk 6 above)
Comment (znarf) [reply components/edit-collective/sections/SetupTwoFactorAuth.js]: The "name" has nothing to do with the API, it's the name of the function that is added to the component.

File: components/edit-collective/sections/SetupTwoFactorAuth.js [hunk 7]
```diff
@@ -307,10 +434,16 @@ const twoFactorAuthToAccountMutation = gqlV2/* GraphQL */ `
   }
 `;
 
-const addTwoFactorAuthToAccountMutation = graphql(twoFactorAuthToAccountMutation, {
-  name: 'addTwoFactorAuthTokenToIndividual',
-  options: { context: API_V2_CONTEXT },
-});
+const addRemoveTwoFactorAuthFromIndividualMutation = gqlV2/* GraphQL */ `
```
Comment (znarf) [inline components/edit-collective/sections/SetupTwoFactorAuth.js:437]: That doesn't not make sense to name this variable "addRemoveTwoFactorAuthFromIndividualMutation", it should be `removeTwoFactorAuthFromIndividualMutation`.

File: components/edit-collective/sections/SetupTwoFactorAuth.js [hunk 8]
```diff
@@ -339,6 +472,16 @@ const addAccountHasTwoFactorAuthData = graphql(accountHasTwoFactorAuthQuery, {
   }),
 });
 
-const addGraphql = compose(addTwoFactorAuthToAccountMutation, addAccountHasTwoFactorAuthData);
+const addGraphql = compose(
+  graphql(addTwoFactorAuthToIndividualMutation, {
+    name: 'addTwoFactorAuthTokenToIndividual',
```
Comment (znarf) [inline components/edit-collective/sections/SetupTwoFactorAuth.js:477]: Ok, looks good.

**Pull request merged**